sympy/assumptions/cnf.py

```diff
@@ -451,3 +508,25 @@ def encode_arg(self, arg):
 
     def encode(self, clause):
         return {self.encode_arg(arg) if not arg.lit == S.false else 0 for arg in clause}
+
+    def decode_literal(self, encoded_arg):
```

Comment:
Can you add tests for these new methods?

Reply:
yep, sure

Reply:
It doesn't look like any tests exist for the EncodedCNF class. In general, all the code in sympy/assumptions/cnf.py lacks tests, though CNF gets tested. (Though CNF and literal do get used in test_satask.py and test_query.py).

I think you should make a new file in sympy/assumptions/tests called "test_cnf" with tests. Don't worry about adding tests for cnf though. Just add tests for EncodedCNF.

Reply:
Actually there are tests for CNF. Bizarrely they're in sympy/logic/tests/test_boolalg.py. 

I would probably add tests to that file rather than creating a whole new file.